From a7b6f23b7fa3f5d1f3ae64034a4aff12fb8c1df0 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Thu, 11 Apr 2024 16:04:44 -0700
Subject: [PATCH] ==== commit cleanup ====

mm/damon: introduce DAMON parameters online commit function

DAMON context struct (damon_ctx) contains user requests (parameters),
internal status, and operation results.  For flexible usages, DAMON API
users are encouraged to manually manipulate the struct.  That works well
for simple use cases.  However, it has turned out that it is not that
simple at least for online parameters udpate.  It is easy to forget
properly maintaining internal status and operation results.  Also, such
manual manipulation for online tuning is implemented multiple times on
DAMON API users including DAMON sysfs interface, DAMON_RECLAIM and
DAMON_LRU_SORT.  As a result, we have multiple sources of bugs for same
problem.  Actually we found and fixed a few bugs from online parameter
updating of DAMON API users.

Implement a function for online DAMON parameters update in core layer,
and replace DAMON API users' manual manipulation code for the use case.
The core layer function could still have bugs, but this change reduces
the source of bugs for the problem to one place.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/RYnTeJnM | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/RYnTeJnM

diff --git a/damon_meta_changes/RYnTeJnM b/damon_meta_changes/RYnTeJnM
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.2

